src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -272,6 +273,10 @@ def _execute_command(kwargs):
         try:
             op = get_op_handler(operation)
             result = op(client, **kwargs) if client else op(**kwargs)
+
+            if expose_no_wait and kwargs.get('raw', None):
+                return None #return None for 'no-wait'
```

Comment:
The `no-wait` is a parameter alias on `raw`. This `if` condition makes sure the command returns nothing on `no-wait`, since we don't have yet the final result to return to users.